File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> curr(128, 0);

        int l = 0;
        int answer = 0;

        for (int r = 0; r < s.size(); r++) {
            curr[s[r]]++;

            while (curr[s[r]] > 1) {
                curr[s[l]]--;
                l++;
            }

            answer = max(answer, r - l + 1);
        }

        return answer;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        answer = nums[0]
        summa = nums[0]

        l = 0
        for r in range(1, len(nums)):
            summa += nums[r]
            while summa < nums[r] and l < r:
                summa -= nums[l]
                l += 1
            # print(l, r, "-", summa)
            answer = max(answer, summa)
        
        return answer